include/libdnf/module/module_package.hpp

```diff
@@ -0,0 +1,222 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2.1 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef LIBDNF_MODULE_MODULE_PACKAGE_HPP
+#define LIBDNF_MODULE_MODULE_PACKAGE_HPP
+
+#include "libdnf/module/module_profile.hpp"
+
+#include <string>
+#include <vector>
+
+struct _ModulemdModuleStream;
+
+
+namespace libdnf::module {
+
+
+struct ModulePackageId {
+public:
+    ModulePackageId() = default;
+    explicit ModulePackageId(int id) : id(id) {}
+
+    bool operator==(const ModulePackageId & other) const noexcept { return id == other.id; }
+    bool operator!=(const ModulePackageId & other) const noexcept { return id != other.id; }
+
+    // Corresponds to solvable id
+    int id{0};
+};
+
+
+// Represents one modulemd document (uniquely described by name:stream:version:context:arch, but there can theoretically be more objects with the same NSVCA)
+// @replaces libdnf:module/ModulePackage.hpp:class:ModulePackage
+class ModulePackage {
+public:
+    /// @return The module name.
+    /// @since 5.0
+    //
+    // @replaces libdnf:module/ModulePackage.hpp:method:ModulePackage.getNameCStr()
+    const char * get_name_cstr() const;
```

Comment:
We need it for a performance reasons internally but do we need it to expose it as API? RPM package does not provide C strings also.